### PR TITLE
Register localusers addressbook in app.php

### PR DESCRIPTION
**Pull request by LEDfan**
@tanghus I have added the registering of the localusers addressbook to the appinfo/app.php

### CODE REVIEW

File: lib/app.php [hunk 1]
```diff
@@ -56,7 +56,7 @@ class App {
 		'ldap' => 'OCA\Contacts\Backend\Ldap',
 		'local' => 'OCA\Contacts\Backend\Database',
 		'shared' => 'OCA\Contacts\Backend\Shared',
-		'localusers' => 'OCA\Contacts\Backend\LocalUsers',
+		//'localusers' => 'OCA\Contacts\Backend\LocalUsers',
```
Comment (LEDfan) [inline lib/app.php:59]: @tanghus is it okay/necessary to delete it here?

**Comment by tanghus** [comment]
@LEDfan I'm afraid we misunderstood each other.

What I meant was to add a method `getSearchProvider()` to `\OCA\Contacts\AddressBook` which returns `$this->getBackend()->getSearchProvider($this);`

Then in `appinfo/app.php` change to:

``` php
    $app = new App($api->getUserId());
    $addressBooks = $app->getAddressBooksForUser();
    foreach ($addressBooks as $addressBook)  {
        if ($addressBook->isActive()) {
            \OCP\Contacts::registerAddressBook($addressBook->getSearchProvider());
        }
    }

```

**Comment by LEDfan** [comment]
@tanghus oh yes indeed. I will update the code.

**Comment by tanghus** [comment]
Maybe check for a result first? Not sure if all backends will implement a searchprovider. Currently only local, shared and localusers are relevant.

**Comment by LEDfan** [comment]
Okay, and then if there isn't one use `\OCA\Contacts\Addressbookprovider` right?

**Comment by tanghus** [comment]
Hmm, no then don't register it.

**Comment by tanghus** [comment]
I've added an empty `getSearchProvider()` method to `AbstractBackend` so it won't barf if a backend hasn't implemented it.

**Comment by LEDfan** [comment]
Replaced by #470 (so I can start from master again)

**Pull request closed without merging**